Baby Monitor/Source Files/AppDependencies.swift

```diff
@@ -41,9 +41,11 @@ final class AppDependencies {
     private(set) lazy var connectionChecker: ConnectionChecker = NetServiceConnectionChecker(netServiceClient: netServiceClient, urlConfiguration: urlConfiguration)
     
     // MARK: - WebRTC
-    
+
+    private(set) lazy var webRtcConnectionProxy: PeerConnectionProxy = RTCPeerConnectionDelegateProxy()
+
     private(set) lazy var webRtcServer: () -> WebRtcServerManagerProtocol = {
-        WebRtcServerManager(peerConnectionFactory: self.peerConnectionFactory)
+        WebRtcServerManager(peerConnectionFactory: self.peerConnectionFactory, connectionDelegateProxy: self.webRtcConnectionProxy)
```

Comment:
passing `self.webRtcConnectionProxy` could potentially be problematic when socket connection is terminated and then re-created. `ConnectionDelegateProxy` is re-created each time `WebSocketWrapper` is re-created, so passing it like that might cause a memory leak and (worse) relying on a `proxy` that is no longer connected to a working `socket`. 
I'd suggest passing `delegate` either wrapped in `LazyClearable` or  passing reference to entire `WebSocket` (wrapped in `LazyClearable` as well) and then get the `delegate` from there

Reply:
// Cleared out during the conversion on slack: this problem won't appear here.

Reply:
I just noticed that it's the Baby App (and RTCServer). You can ignore the comment above, as server instance is not clearable throughout child app lifetime